test/multiverse/suites/grpc/grpc_server_test.rb

```diff
@@ -39,6 +39,37 @@ def method
     m
   end
 
+  def current_segment
+    t = MiniTest::Mock.new
+    t.expect(:add_agent_attribute, nil, [:'request.headers', {}])
+    t.expect(:add_agent_attribute, nil, [:'request.uri', "grpc://:/"])
+    t.expect(:add_agent_attribute, nil, [:'request.method', nil])
+    t.expect(:add_agent_attribute, nil, [:'request.grpc_type', nil])
+    t
+  end
+
+  def destinations
+    NewRelic::Agent::Instrumentation::GRPC::Server::DESTINATIONS
+  end
+
+  def transaction
+    t = MiniTest::Mock.new
+    t.expect(:add_agent_attribute, nil, [:'request.headers', {}, destinations])
+    t.expect(:add_agent_attribute, nil, [:'request.uri', "grpc://:/", destinations])
+    t.expect(:add_agent_attribute, nil, [:'request.method', nil, destinations])
+    t.expect(:add_agent_attribute, nil, [:'request.grpc_type', nil, destinations])
+    t.expect(:current_segment, nil) # 4 existence checks
+    t.expect(:current_segment, nil)
+    t.expect(:current_segment, nil)
+    t.expect(:current_segment, nil)
+    t.expect(:current_segment, current_segment) # 4 attrs to add
+    t.expect(:current_segment, current_segment)
+    t.expect(:current_segment, current_segment)
+    t.expect(:current_segment, current_segment)
+    t.expect(:finish, nil)
+    t
+  end
```

Comment:
Mocks and I are still getting familiar with each other. Is `t.expect` an assertion? Does it behave like RSpec's `expect`? I'd love some guidance on how these are used!

Reply:
Yes, MiniTest's `allow` vs `expect` (permit vs. require) is similar. I believe the RSpec equivalent of this code:

```ruby
t.expect(:current_segment, nil) # 4 existence checks
t.expect(:current_segment, nil)
t.expect(:current_segment, nil)
t.expect(:current_segment, nil)
```

would be:

```ruby
# note: and_return(nil) would be redundant with RSpec's default behavior
expect(t).to receive(:current_segment).exactly(4).times
```

and these:

```ruby
t.expect(:add_agent_attribute, nil, [:'request.headers', {}, destinations])
t.expect(:add_agent_attribute, nil, [:'request.uri', "grpc://:/", destinations])
t.expect(:add_agent_attribute, nil, [:'request.method', nil, destinations])
t.expect(:add_agent_attribute, nil, [:'request.grpc_type', nil, destinations])
```

would translate to:

```ruby
# note: and_return(nil) would be redundant with RSpec's default behavior
expect(t).to receive(:add_agent_attribute).with([:'request.headers', {}, destinations]).exactly(4).times
```

Reply:
Thank you!